src/set_theory/ordinal/notation.lean

```diff
@@ -791,6 +791,142 @@ begin
       exact (repr_opow_aux₂ _ ad a00 al _ _).2 } }
 end
 
+/-- Given an ordinal, returns `inl none` for `0`, `inl (some a)` for `a+1`, and
+  `inr f` for a limit ordinal `a`, where `f i` is a sequence converging to `a`. -/
+def fundamental_seq : onote → option onote ⊕ (ℕ → onote)
+| zero := sum.inl none
+| (oadd a m b) :=
+  match fundamental_seq b with
+  | sum.inr f := sum.inr (λ i, oadd a m (f i))
+  | sum.inl (some b') := sum.inl (some (oadd a m b'))
+  | sum.inl none := match fundamental_seq a, m.nat_pred with
+    | sum.inl none, 0 := sum.inl (some zero)
+    | sum.inl none, m+1 := sum.inl (some (oadd zero m.succ_pnat zero))
+    | sum.inl (some a'), 0 := sum.inr (λ i, oadd a' i.succ_pnat zero)
+    | sum.inl (some a'), m+1 := sum.inr (λ i, oadd a m.succ_pnat (oadd a' i.succ_pnat zero))
+    | sum.inr f, 0 := sum.inr (λ i, oadd (f i) 1 zero)
+    | sum.inr f, m+1 := sum.inr (λ i, oadd a m.succ_pnat (oadd (f i) 1 zero))
+    end
+  end
+
+theorem exists_lt_add {α} [hα : nonempty α] {o : ordinal} {f : α → ordinal}
+  (H : ∀ a, a < o → ∃ i, a < f i) (b : ordinal) (a) (h : a < b + o) : ∃ i, a < b + f i :=
```

Comment:
It is, but there was no easy way to apply it to get this theorem. This particular statement is preferred over alternatives because it is what is needed in the proof.